lib/moonshot/build_mechanism/github_release.rb

```diff
@@ -170,6 +169,27 @@ def fetch_changes(version)
       end
     end
 
+    # Checks for the commit's CI job status. If its not finished yet,
+    # wait till timeout.
+    #
+    # @param sha [String] Commit sha.
+    #
+    # @return [String] Status and links to the CI jobs
+    def check_ci_status(sha)
+      out = nil
+      ilog.start_threaded('Validate `CHANGELOG.md`.') do |step|
+        Retriable.retriable on: RuntimeError,
```

Comment:
btw are you guys talking about a default retry method to be used here or modifying sh_out in shell.rb to support retry as I suggested?